src/sentry/snuba/metrics/extraction.py

```diff
@@ -49,6 +49,13 @@
 
 logger = logging.getLogger(__name__)
 
+# This helps us control the different spec versions
+# in order to migrate customers from invalid specs
+SPEC_VERSIONS = [
+    {"use_updated_env_logic": True},  # The latest version
```

Comment:
Is this the default we should have if no spec_version is passed?

Reply:
In my opinion, a default should be the V0 which has no flags.

How I envisioned the specs versioning system would be something like:
```python
MIN_VERSION = 1

VERSIONS = [
	Version(0),
	Version(1, ["flag_1", "flag_2"]),
	... # add new versions here
]

def get_spec_versions(min_version = 1) -> Sequence[Version]:
	return filter_versions(min_version)

...

for version in get_spec_versions():
	if version.has("flag_1"):
		# this something
```

With this system, we should not disable past versions by removing them but rather specify a minimum version so that we evolve with the history persisted. This can be useful for example when removing flags for cleanup, we can just create a new version without that flag and update the minimum version. Keeping the system version with a simple integer makes it much easier to debug since you know that you will have from version `x` to version `y` which generate specs with different params.

Reply:
The main rationale behind using `Version` objects is that they can encapsulate the behavior of fetching tags and return false when a tag is not there. Quite simple and doesn't require any duplication.

Reply:
However, having a simple dictionary and leveraging the order of the list to define versioning is semantically identical but not very explicit and ergonomic in my opinion.